demo/scene_emibl.hpp

```diff
@@ -95,107 +95,107 @@ namespace emibl_scene
 
 			{
 				// Create Material
-				rusty_metal_material = material_pool->Create();
+				rusty_metal_material = material_pool->Create(texture_pool.get());
 
 				wr::Material* rusty_metal_internal = material_pool->GetMaterial(rusty_metal_material);
 
-				rusty_metal_internal->SetAlbedo(metal_splotchy_albedo);
-				rusty_metal_internal->SetNormal(metal_splotchy_normal);
-				rusty_metal_internal->SetRoughness(metal_splotchy_roughness);
-				rusty_metal_internal->SetMetallic(metal_splotchy_metallic);
+				rusty_metal_internal->SetTexture(wr::MaterialTextureType::ALBEDO, metal_splotchy_albedo);
```

Comment:
Why remove the explicit calls to setting the type of material? "SetAlbedo" is much more clear than calling SetTexture(MaterialTextureType::ALBEDO. Consider using more explicit naming

Reply:
It leaves a lot of duplicated code, makes it harder to add an additional texture type. Now you just add to the enum.